src/test/java/com/salessparrow/api/unit/repositories/SalesforceUserRepositoryTest.java

```diff
@@ -0,0 +1,125 @@
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.github.dynamobee.exception.DynamobeeException;
+import com.salessparrow.api.domain.SalesforceUser;
+import com.salessparrow.api.exception.CustomException;
+import com.salessparrow.api.helper.Cleanup;
+import com.salessparrow.api.helper.Common;
+import com.salessparrow.api.helper.FixtureData;
+import com.salessparrow.api.helper.LoadFixture;
+import com.salessparrow.api.helper.Setup;
+import com.salessparrow.api.lib.errorLib.ErrorObject;
+import com.salessparrow.api.repositories.SalesforceUserRepository;
+
+@SpringBootTest
+@Import({ Setup.class, Cleanup.class, Common.class, LoadFixture.class })
+public class SalesforceUserRepositoryTest {
+
+    @Autowired
+    private Setup setup;
+
+    @Autowired
+    private Cleanup cleanup;
+
+    @Autowired
+    private Common common;
+
+    @Autowired
+    private LoadFixture loadFixture;
+
+    @BeforeEach
+    public void setUp() throws DynamobeeException, IOException {
+        setup.perform();
+    }
+
+    @AfterEach
+    public void tearDown() {
+        cleanup.perform();
+    }
+
+    @Autowired
+    private DynamoDBMapper dynamoDBMapper;
+
+        @Test
+        public void testSaveSalesforceUser() {
+            //Valid Save Db Query
+            SalesforceUser salesforceUserValid = new SalesforceUser();
+            salesforceUserValid.setExternalUserId("externalUserId-test1");
+            SalesforceUserRepository salesforceUserRepository = new SalesforceUserRepository(dynamoDBMapper);
+            SalesforceUser salesforceUserResp = salesforceUserRepository.saveSalesforceUser(salesforceUserValid);
+            assertEquals(salesforceUserValid.getExternalUserId(), salesforceUserResp.getExternalUserId());
+
+            // Invalid Save Db Query without partition key
+            SalesforceUser salesforceUserInvalid = new SalesforceUser();
+            salesforceUserInvalid.setExternalUserId("externalUserId-test2");
+            
+            // mock the DynamoDBMapper
+            DynamoDBMapper dynamoDBMapper = mock(DynamoDBMapper.class);
+            SalesforceUserRepository mockSalesforceUserRepository = new SalesforceUserRepository(dynamoDBMapper);
+
+            // Mock the behavior to throw an exception when save is called
+            doThrow(new CustomException(new ErrorObject("test:r_sotr_tssfu_1", "something_went_wrong", "mock db save error")))
+            .when(dynamoDBMapper)
+            .save(salesforceUserInvalid);
+
+            // Test if CustomException is thrown with the expected error code
+            CustomException thrownException = assertThrows(
+                CustomException.class, 
+                () -> mockSalesforceUserRepository.saveSalesforceUser(salesforceUserInvalid)
+            );
+            // Validate the error identifier to be a 500 error
+            assertEquals("something_went_wrong", thrownException.getErrorObject().getApiErrorIdentifier());
```

Comment:
The error is a generic 500 error instead of a specific error as it is unexpected. But for known errors the generic "something_went_wrong" error config should not be used